templates/console/source/lambda/config-version/cf_config_version_manager_graphql/cf_config_version_manager_graphql.py

```diff
@@ -656,33 +707,14 @@
     ddb_client = boto3.resource('dynamodb')
     ddb_table = ddb_client.Table(DDB_SNAPSHOT_TABLE_NAME)
 
-    response = ddb_table.get_item(
-        Key={
-            "distributionId": distributionId,
-            "snapShotName": snapShotName
-        })
-    logger.info(response)
-    if 'Item' in response:
-        # No duplicate snapShotName allowed
-        raise Exception("There is already snapShotName:" + snapShotName)
+    check_snapshot_name_duplication(ddb_table, distributionId, snapShotName)
 
-    # create a record in snapshot ddb table with the snapShotName
-    # first get the latest versionId of updated distribution
-    latest_table = ddb_client.Table(DDB_LATESTVERSION_TABLE_NAME)
-    try:
-        resp = latest_table.query(
-            KeyConditionExpression=Key('distributionId').eq(distributionId)
-        )
-        log.info(resp)
-    except ClientError as e:
-        logging.error(e)
+    latest_version = get_latest_version(ddb_client, distributionId)
 
-    record_list = resp['Items']
-    if len(record_list) == 0:
-        raise Exception("There is no latest version for distributionId:" + distributionId)
-    latest_version = record_list[0]['versionId']
-    logging.info("The latest version of distribution:" + str(distributionId) + " is " + str(latest_version))
+    return create_snapshot(ddb_table, distributionId, latest_version, snapShotName, snapShotNote)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;snapShotNote&#39; (index: 2 | type: str). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;snapShotName&#39;, &#39;distributionId&#39;. The same parameter type is validated here for example: aws-cloudfront-extensions/templates/console/source/lambda/config-version/cf_config_version_manager_graphql/cf_config_version_manager_graphql.py:196. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness, emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles) using single or double underscore.

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).